iceberg/iceberg-handler/src/main/java/org/apache/iceberg/data/PartitionStatsHandler.java

```diff
@@ -0,0 +1,336 @@
+   * @return a schema that corresponds to the provided unified partition type.
+   */
+  public static Schema schema(StructType partitionType) {
+    Preconditions.checkState(!partitionType.fields().isEmpty(), "table must be partitioned");
+
+    return new Schema(
+      NestedField.required(1, Column.PARTITION.name(), partitionType),
+      NestedField.required(2, Column.SPEC_ID.name(), IntegerType.get()),
+      NestedField.required(3, Column.DATA_RECORD_COUNT.name(), LongType.get()),
+      NestedField.required(4, Column.DATA_FILE_COUNT.name(), IntegerType.get()),
+      NestedField.required(5, Column.TOTAL_DATA_FILE_SIZE_IN_BYTES.name(), LongType.get()),
+      NestedField.optional(6, Column.POSITION_DELETE_RECORD_COUNT.name(), LongType.get()),
+      NestedField.optional(7, Column.POSITION_DELETE_FILE_COUNT.name(), IntegerType.get()),
+      NestedField.optional(8, Column.EQUALITY_DELETE_RECORD_COUNT.name(), LongType.get()),
+      NestedField.optional(9, Column.EQUALITY_DELETE_FILE_COUNT.name(), IntegerType.get()),
+      NestedField.optional(10, Column.TOTAL_RECORD_COUNT.name(), LongType.get()),
+      NestedField.optional(11, Column.LAST_UPDATED_AT.name(), LongType.get()),
+      NestedField.optional(12, Column.LAST_UPDATED_SNAPSHOT_ID.name(), LongType.get()));
+  }
+
+  /**
+   * Computes and writes the {@link PartitionStatisticsFile} for a given table's current snapshot.
+   *
+   * @param table The {@link Table} for which the partition statistics is computed.
+   * @return {@link PartitionStatisticsFile} for the current snapshot.
+   */
+  public static PartitionStatisticsFile computeAndWriteStatsFile(Table table) {
+    return computeAndWriteStatsFile(table, null);
+  }
+
+  /**
+   * Computes and writes the {@link PartitionStatisticsFile} for a given table and branch.
+   *
+   * @param table The {@link Table} for which the partition statistics is computed.
+   * @param branch A branch information to select the required snapshot.
+   * @return {@link PartitionStatisticsFile} for the given branch.
+   */
+  public static PartitionStatisticsFile computeAndWriteStatsFile(Table table, String branch) {
+    Snapshot currentSnapshot = SnapshotUtil.latestSnapshot(table, branch);
+    if (currentSnapshot == null) {
+      Preconditions.checkArgument(
+          branch == null, "Couldn't find the snapshot for the branch %s", branch);
+      return null;
+    }
+
+    StructType partitionType = Partitioning.partitionType(table);
+    Schema schema = schema(partitionType);
+
+    Collection<PartitionStats> stats = PartitionStatsUtil.computeStats(table, currentSnapshot);
+    List<PartitionStats> sortedStats = PartitionStatsUtil.sortStats(stats, partitionType);
+    Iterator<PartitionStatsRecord> convertedRecords = statsToRecords(sortedStats, schema);
+    return writePartitionStatsFile(table, currentSnapshot.snapshotId(), schema, convertedRecords);
+  }
+
+  @VisibleForTesting
+  static PartitionStatisticsFile writePartitionStatsFile(
```

Comment:
This method is not called from unit tests, @VisibleForTesting seems to be not needed.

Reply:
would be removed once we upgrade iceberg dep

Reply:
removed